impl/src/main/java/org/apache/myfaces/component/search/SearchExpressionHandlerImpl.java

```diff
@@ -534,8 +534,8 @@ public void invokeContextCallback(FacesContext context, UIComponent target)
             // At this point if the algorithm hasn't returned and the topExpression does not have any separator char
             // we need to do the search backward using findComponent.
             if (target == null 
-                    && searchExpressionContext.getSource() == previous 
-                    && (topExpression.indexOf(separatorChar) == -1) )
+                    && searchExpressionContext.getSource() == previous )
+                    // && (topExpression.indexOf(separatorChar) == -1) ) removed for MYFACES-4695
```

Comment:
if you replace it with 
`&& (topExpression.indexOf(":@") == -1))`

im fine with it; please dont forget all other branches